platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc

```diff
@@ -140,6 +140,71 @@ java {
 ======
 =====
 
+[[publishing_artifact_name_different_from_artifact_id_maven]]
+==== Deprecated publishing artifact dependencies with explicit name to Maven repositories
+
+Publishing dependencies with an explicit artifact with a name different from the dependency's `artifactId` to Maven repositories has been deprecated.
+This behavior is still permitted when publishing to Ivy repositories.
+It will result in an error in Gradle 9.0.
```

Comment:
In the future, will this become an error or will we produce a POM that depends on `org:foo:1.0`?

Reply:
This will become an error. An artifact with name != artifactId is different than a dependency on just the artifactId. We should not ignore the declared name during publishing and therefore should throw an error